common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -542,3 +544,117 @@ def test_courseware_a11y(self):
+    problem2_name = 'MULTIPLE CHOICE TEST PROBLEM 2'
+
+    def setUp(self):
+        super(ProblemStateOnNavigationTest, self).setUp()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        # Install a course with section, tabs and multiple choice problems.
+        course_fix = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section 1').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection 1,1').add_children(
+                    self.get_multiple_choice_problem(self.problem1_name),
+                    self.get_multiple_choice_problem(self.problem2_name),
+                ),
+            ),
+        ).install()
+
+        # Auto-auth register for the course.
+        AutoAuthPage(
+            self.browser, username=self.USERNAME, email=self.EMAIL,
+            course_id=self.course_id, staff=False
+        ).visit()
+
+        self.courseware_page.visit()
+        self.problem_page = ProblemPage(self.browser)
+
+    def get_multiple_choice_problem(self, problem_name):
+        """
+        Return the Multiple Choice Problem Descriptor, given the name of the problem.
+        """
+        factory = MultipleChoiceResponseXMLFactory()
+        xml_data = factory.build_xml(
+            question_text='The correct answer is Choice 2',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+
+        return XBlockFixtureDesc(
+            'problem',
+            problem_name,
+            data=xml_data,
+            metadata={'rerandomize': 'always'}
+        )
+
+    @ddt.data(
+        ('check', 'choice_choice_1', 'label.choicegroup_incorrect'),
+        ('reset', 'choice_choice_1', ['label.choicegroup_incorrect', 'span.unanswered']),
+        ('save', 'choice_choice_1', 'div.capa_alert'),
+    )
+    @ddt.unpack
+    def test_perform_problem_actions_and_navigate(self, action, selected_choice, status_selector):
+        # Scenario:
+        # I go to sequential position 1
+        # Facing problem1, I select 'choice_1'
+        # Then perform the action – check/save/reset
+        # Then I go to sequential position 2
+        # Then I came back to sequential position 1 again
+        # Facing problem1, I observe the problem1 content is not
+        # outdated before and after sequence navigation
+
+        # Go to sequential position 1 and assert that we are on problem 1.
+        self.courseware_page.go_to_sequential_position(1)
+        self.assertEquals(self.problem_page.problem_name, self.problem1_name)
+
+        # Update problem 1's content state – by performing check/reset/save operation.
+        self.problem_page.click_choice(selected_choice)
+        if action == 'check':
```

Comment:
It's not going to be super DRY, but I think it makes more sense to split this up in to 3 test cases. You can abstract out helper functions if needed, but it doesn't feel like `ddt` is the right move here. (In my mind, the test for whether you should use `ddt` or split test cases is if you need an `if` statement or not. If you need `if`, you need a new test case.)